frontend/src/Components/AlgInputs/AlgInputsCBA.js

```diff
@@ -114,16 +115,41 @@ class AlgInputsCBA extends Component {
 
         // if the CBA input relationship doesn't exist, insert new CBA input table rows to db
         if(!config_res.data.length){
-            const postUrl = `${ serverUrl }/cost_benefit_analysis_runner`;
+
+            // starts progress bar to show loading to user
+            this.props.loadingResults(true);
+
+            // data inputs to run the cba tool
             const profileMatch = this.state.profileData.filter((profile) => profile.config_name === this.state.profileName);
             const countyMatch = profileMatch.map(profile => profile["choice"]);
-            axios({
+
+            // runs cba task with celery on backend
+            const cba_res = await axios({
+                url: `${ serverUrl }/cost_benefit_analysis_runner`,
                 method: "post",
-                url: postUrl,
-                data: {load_profile: this.state.profileName, county: countyMatch},
+                data: {load_profile: this.state.profileName, county: countyMatch}
+            
             });
+
+            // celery-flower monitoring to check task status on cba tool
+            let status = cba_res.data.status;
+            const task_id = cba_res.data.task_id;
+            while (status !== "SUCCESS") {
```

Comment:
There's an assumption here that the task will always succeed. 
I think if the task fails, we should stop the loop. I also think we should implement a simple exponential backoff algorithm, something like: https://jsfiddle.net/pajtai/pLka0ow9

Reply:
fixed